lib/internal/event_target.js

```diff
@@ -362,13 +369,21 @@ class Listener {
       previous.next = this;
     this.previous = previous;
     this.listener = listener;
-    // TODO(benjamingr) these 4 can be 'flags' to save 3 slots
-    this.once = once;
-    this.capture = capture;
-    this.passive = passive;
-    this.isNodeStyleListener = isNodeStyleListener;
+
+    let flags = 0b0;
+    if (once)
+      flags |= kFlagOnce;
+    if (capture)
+      flags |= kFlagCapture;
+    if (passive)
+      flags |= kFlagPassive;
+    if (isNodeStyleListener)
+      flags |= kFlagNodeStyle;
+    if (weak)
+      flags |= kFlagWeak;
+    this.flags = flags;
```

Comment:
Really not found of this, I find it harder to read, and my guess is that it's much less performant to reassign the same variable several time rather than once.

Reply:
@aduh95 there are several good tools for this, the easiest way is probably to run with `--log-all` and use the syste analyzer https://v8.github.io/tools/head/system-analyzer/index.html

Reply:
(If you really just want to see the bytecode emitted then --print-bytecode would also work)